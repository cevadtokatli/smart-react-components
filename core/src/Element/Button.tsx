import styled from 'styled-components'
import applyStyledProps, { StyledProps, shouldForwardProp } from '../styled-props'

export default styled.button.withConfig({ shouldForwardProp })<StyledProps>(applyStyledProps)
