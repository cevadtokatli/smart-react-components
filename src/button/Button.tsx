import React from "react"
import Props from "./Props"
import DV from "../default-value"
import IconElement from "./IconElement"
import ButtonElement from "./Element"
import Loading from "./Loading"
import WaveEffect from "../wave-effect/WaveEffect"

const Button: React.FC<Props> = ({size="default",sizeSm,sizeMd,sizeLg,sizeXl,elementProps=DV.JSX_ELEMENT_PROPS,children,type,fill,link,shape,block,fixedSize,waveEffect="light",icon,iconPosition="left",loading,loadingStatus,loadingTransitionClassName="src-button-loading",loadingTransitionType="ease-in-out",loadingTransitionDuration=300,loadingTransitionShowAnimation=true,loadingTransitionHideAnimation=true}) => {
    const IconEl = icon ? <IconElement data-src-icon-position={iconPosition}>{icon}</IconElement> : null
     
    const ButtonEl = (
        <ButtonElement {...elementProps} size$={size} size$Sm={sizeSm} size$Md={sizeMd} size$Lg={sizeLg} size$Xl={sizeXl} type$={type} fill$={fill} link={link} shape={shape} block={block} fixedSize={fixedSize} loading$={loadingStatus}>
            {loading && <Loading element={loading} status={loadingStatus} transitionClassName={loadingTransitionClassName} transitionDuration={loadingTransitionDuration} transitionType={loadingTransitionType} transitionShowAnimation={loadingTransitionShowAnimation} transitionHideAnimation={loadingTransitionHideAnimation} />}
            {iconPosition == "left" && IconEl}
            {(!loading || typeof children !== "string") ? children : <span>{children}</span>}
            {iconPosition == "right" && IconEl}
        </ButtonElement>
    )

    return waveEffect ? <WaveEffect type={waveEffect}>{ButtonEl}</WaveEffect> : ButtonEl
}

export {default as Props} from "./Props"
export default Button