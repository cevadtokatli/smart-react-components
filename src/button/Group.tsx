import React from "react"
import {SizeProps} from "../props/size-props"
import {JSXElementProps} from "../props"
import DV from "../default-value"
import GroupElement from "./GroupElement"

export interface Props extends SizeProps {
    children: JSX.Element[]
    elementProps?: JSXElementProps
    type?: string
    fill?: boolean
    shape?: string
    fixedSize?: boolean
}

const ButtonGroup: React.FC<Props> = ({size="default",sizeSm,sizeMd,sizeLg,sizeXl,children,elementProps=DV.JSX_ELEMENT_PROPS,type="primary",fill=true,shape="default",fixedSize}) => {
    return (
        <GroupElement {...elementProps}>
            {children.map((item, index) => React.cloneElement(item, {key: item.key || index,size,sizeSm,sizeMd,sizeLg,sizeXl,type,fill,shape,fixedSize}))}
        </GroupElement>
    )
}

export default ButtonGroup