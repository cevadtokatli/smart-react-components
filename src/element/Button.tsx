import styled from "styled-components"
import {Props, attrs, css} from "./"

export default styled.button.attrs<Props>(attrs)<Props>(css)