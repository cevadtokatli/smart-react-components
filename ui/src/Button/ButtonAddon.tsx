import extractElementProps from '@smart-react-components/core/element-props'
import clickEvents, { ClickEvents } from '@smart-react-components/core/element-props/click-events'
import { ContentElement, PaletteProp, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import React from 'react'
import { OrderPosition } from '../types'
import WaveEffect from '../WaveEffect'
import ButtonAddonElement from '../components/Button/ButtonAddonElement'
import { getWaveEffectPalette } from '../util/wave-effect'

export interface Props extends
  Partial<ResponsiveProp<'size', SizeProp>>,
  ClickEvents {
  children: ContentElement
  isDisabled?: boolean
  isOutline?: boolean
  isSeparated?: boolean
  isSoft?: boolean
  palette?: PaletteProp
  waveEffectPalette?: PaletteProp
}

interface PrivateProps {
  hasHover: boolean
  hasWaveEffect: boolean
  position: OrderPosition
  shape: ShapeProp
}

const ButtonAddon: React.FC<Props> = (props: Props & PrivateProps) => {
  const El = (
    <ButtonAddonElement
      addonPosition={props.position}
      buttonSize={props.size}
      buttonSizeSm={props.sizeSm}
      buttonSizeMd={props.sizeMd}
      buttonSizeLg={props.sizeLg}
      buttonSizeXl={props.sizeXl}
      hasHover={props.hasHover}
      isDisabled={props.isDisabled}
      isOutline={props.isOutline}
      isSeparated={props.isSeparated}
      isSoft={props.isSoft}
      palette={props.palette}
      shape={props.shape}
      {...(!props.isDisabled && extractElementProps(props, [clickEvents]))}
    >
      {props.children}
    </ButtonAddonElement>
  )

  return props.isSeparated && props.hasWaveEffect ? <WaveEffect palette={getWaveEffectPalette(props.waveEffectPalette, props.palette, props.isSoft)}>{El}</WaveEffect> : El
}

ButtonAddon.displayName = 'SRCAddon'

export default ButtonAddon
