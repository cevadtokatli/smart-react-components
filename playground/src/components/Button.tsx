//import { ButtonIcon, ButtonList } from '@smart-react-components/ui'
import { Div } from '@smart-react-components/core'
import { Option, OptionGroup } from '@smart-react-components/ui'
import CloseIcon from '@smart-react-components/ui/Alert/AlertCloseIcon'
import Button from '@smart-react-components/ui/Button'
import ButtonLink from '@smart-react-components/ui/Button/ButtonLink'
import ButtonSelect from '@smart-react-components/ui/Button/ButtonSelect'
import React from 'react'

export const Palette = () => (
  <Div display="flex">
    <Button marginRight="$length.3" marginBottom="$length.3">Primary</Button>
    <Button palette="secondary" marginRight="$length.3" marginBottom="$length.3">Secondary</Button>
    <Button palette="success" marginRight="$length.3" marginBottom="$length.3">Success</Button>
    <Button palette="danger" marginRight="$length.3" marginBottom="$length.3">Danger</Button>
    <Button palette="warning" marginRight="$length.3" marginBottom="$length.3">Warning</Button>
    <Button palette="info" marginRight="$length.3" marginBottom="$length.3">Info</Button>
    <Button palette="dynamic" marginRight="$length.3" marginBottom="$length.3">Dynamic</Button>
    <Button palette="!dynamic" marginRight="$length.3" marginBottom="$length.3">!Dynamic</Button>
  </Div>
)

export const Soft = () => (
  <Div display="flex">
    <Button isSoft marginRight="$length.3" marginBottom="$length.3">Primary</Button>
    <Button palette="secondary" isSoft marginRight="$length.3" marginBottom="$length.3">Secondary</Button>
    <Button palette="success" isSoft marginRight="$length.3" marginBottom="$length.3">Success</Button>
    <Button palette="danger" isSoft marginRight="$length.3" marginBottom="$length.3">Danger</Button>
    <Button palette="warning" isSoft marginRight="$length.3" marginBottom="$length.3">Warning</Button>
    <Button palette="info" isSoft marginRight="$length.3" marginBottom="$length.3">Info</Button>
    <Button palette="dynamic" isSoft marginRight="$length.3" marginBottom="$length.3">Dynamic</Button>
    <Button palette="!dynamic" isSoft marginRight="$length.3" marginBottom="$length.3">!Dynamic</Button>
  </Div>
)

export const Outline = () => (
  <Div display="flex">
    <Button isOutline marginRight="$length.3" marginBottom="$length.3">Primary</Button>
    <Button palette="secondary" isOutline marginRight="$length.3" marginBottom="$length.3">Secondary</Button>
    <Button palette="success" isOutline marginRight="$length.3" marginBottom="$length.3">Success</Button>
    <Button palette="danger" isOutline marginRight="$length.3" marginBottom="$length.3">Danger</Button>
    <Button palette="warning" isOutline marginRight="$length.3" marginBottom="$length.3">Warning</Button>
    <Button palette="info" isOutline marginRight="$length.3" marginBottom="$length.3">Info</Button>
    <Button palette="dynamic" isOutline marginRight="$length.3" marginBottom="$length.3">Dynamic</Button>
    <Button palette="!dynamic" isOutline marginRight="$length.3" marginBottom="$length.3">!Dynamic</Button>
  </Div>
)

export const OutlineSoft = () => (
  <Div display="flex">
    <Button isOutline isSoft marginRight="$length.3" marginBottom="$length.3">Primary</Button>
    <Button palette="secondary" isOutline isSoft marginRight="$length.3" marginBottom="$length.3">Secondary</Button>
    <Button palette="success" isOutline isSoft marginRight="$length.3" marginBottom="$length.3">Success</Button>
    <Button palette="danger" isOutline isSoft marginRight="$length.3" marginBottom="$length.3">Danger</Button>
    <Button palette="warning" isOutline isSoft marginRight="$length.3" marginBottom="$length.3">Warning</Button>
    <Button palette="info" isOutline isSoft marginRight="$length.3" marginBottom="$length.3">Info</Button>
    <Button palette="dynamic" isOutline isSoft marginRight="$length.3" marginBottom="$length.3">Dynamic</Button>
    <Button palette="!dynamic" isOutline isSoft marginRight="$length.3" marginBottom="$length.3">!Dynamic</Button>
  </Div>
)

export const Link = () => (
  <Div display="flex">
    <Button isLink marginRight="$length.3" marginBottom="$length.3">Primary</Button>
    <Button palette="secondary" isLink marginRight="$length.3" marginBottom="$length.3">Secondary</Button>
    <Button palette="success" isLink marginRight="$length.3" marginBottom="$length.3">Success</Button>
    <Button palette="danger" isLink marginRight="$length.3" marginBottom="$length.3">Danger</Button>
    <Button palette="warning" isLink marginRight="$length.3" marginBottom="$length.3">Warning</Button>
    <Button palette="info" isLink marginRight="$length.3" marginBottom="$length.3">Info</Button>
    <Button palette="dynamic" isLink marginRight="$length.3" marginBottom="$length.3">Dynamic</Button>
    <Button palette="!dynamic" isLink marginRight="$length.3" marginBottom="$length.3">!Dynamic</Button>
  </Div>
)

export const Shape = () => (
  <Div display="flex">
    <Button marginRight="$length.3" marginBottom="$length.3">Rectangle</Button>
    <Button shape="rounded" marginRight="$length.3" marginBottom="$length.3">Rounded</Button>
  </Div>
)

export const Size = () => (
  <Div display="flex" alignItems="flex-start">
    <Button size="small" marginRight="$length.3" marginBottom="$length.3">Small</Button>
    <Button marginRight="$length.3" marginBottom="$length.3">Medium</Button>
    <Button size="large" marginRight="$length.3" marginBottom="$length.3">Large</Button>
    <Button isBlock>Block</Button>
  </Div>
)

export const Circle = () => (
  <Div>
    <Button isFixedSize shape="rounded" marginRight="$length.3" marginBottom="$length.3">1</Button>
  </Div>
)

export const ButtonLinkC = () => (
  <Div>
    <ButtonLink to="/primary" marginRight="$length.3" marginBottom="$length.3">Primary</ButtonLink>
    <ButtonLink to="/secondary" palette="secondary" marginRight="$length.3" marginBottom="$length.3">Secondary</ButtonLink>
    <ButtonLink to="/success" palette="success" marginRight="$length.3" marginBottom="$length.3">Success</ButtonLink>
    <ButtonLink to="/danger" palette="danger" marginRight="$length.3" marginBottom="$length.3">Danger</ButtonLink>
    <ButtonLink to="/warning" palette="warning" marginRight="$length.3" marginBottom="$length.3">Warning</ButtonLink>
    <ButtonLink to="/info" palette="info" marginRight="$length.3" marginBottom="$length.3">Info</ButtonLink>
    <ButtonLink to="/dynamic" palette="dynamic" marginRight="$length.3" marginBottom="$length.3">Dynamic</ButtonLink>
    <ButtonLink to="/!dynamic" palette="!dynamic" marginRight="$length.3" marginBottom="$length.3">!Dynamic</ButtonLink>
  </Div>
)

export const ButtonSelectC = () => {
  const [active, setActive] = React.useState(() => [])

  return (
    <>
      <ButtonSelect active={active} setActive={setActive} placeholder="Primary">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </ButtonSelect>
      <ButtonSelect palette="secondary" active={active} setActive={setActive} placeholder="Secondary">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </ButtonSelect>
      <ButtonSelect palette="success" active={active} setActive={setActive} placeholder="Success">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </ButtonSelect>
      <ButtonSelect palette="danger" active={active} setActive={setActive} placeholder="Danger">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </ButtonSelect>
      <ButtonSelect palette="warning" active={active} setActive={setActive} placeholder="Warning">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </ButtonSelect>
      <ButtonSelect palette="info" active={active} setActive={setActive} placeholder="Info">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </ButtonSelect>
      <ButtonSelect palette="dynamic" active={active} setActive={setActive} placeholder="Dynamic">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </ButtonSelect>
      <ButtonSelect palette="!dynamic" active={active} setActive={setActive} placeholder="!Dynamic">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </ButtonSelect>
    </>
  )
}