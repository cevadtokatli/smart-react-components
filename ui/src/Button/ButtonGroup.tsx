import React from 'react'
import ButtonGroupElement from '../components/Button/ButtonGroupElement'
import { ButtonListGenericProps } from '../types/button'
import ButtonList from './ButtonList'

export interface Props extends ButtonListGenericProps {}

const ButtonGroup: React.FC<Props> = props => (
  <ButtonList
    {...props}
    hasSpace={false}
    template={<ButtonGroupElement shape={props.shape} />}
  >
    {props.children}
  </ButtonList>
)

ButtonGroup.defaultProps = {
  shape: 'rounded',
}

export default ButtonGroup
