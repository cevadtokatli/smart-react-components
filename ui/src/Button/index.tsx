import Div from '@smart-react-components/core/Element/Div'
import extractElementProps from '@smart-react-components/core/element-props'
import clickEvents, { ClickEvents } from '@smart-react-components/core/element-props/click-events'
import intrinsicStyledCoreProps from '@smart-react-components/core/element-props/intrinsic-styled-core-props'
import useChangeEffect from '@smart-react-components/core/hooks/useChangeEffect'
import { ContentElement, Nullable } from '@smart-react-components/core/types'
import React from 'react'
import { StyledComponent, ThemeContext } from 'styled-components'
import ButtonAddon from './ButtonAddon'
import ButtonLoading from './ButtonLoading'
import ButtonElement from '../components/Button/ButtonElement'
import ButtonContent from '../components/Button/ButtonContent'
import useAddons from '../hooks/useAddons'
import { getWaveEffectPalette } from '../util/wave-effect'
import { ButtonGenericProps, ButtonType } from '../types/button'
import WaveEffect from '../WaveEffect'

export interface Props extends
  ButtonGenericProps<ContentElement>,
  ClickEvents {
  as?: string | StyledComponent<any, any>
  type?: ButtonType
}

const Button = React.forwardRef<HTMLElement, Props>((props, forwardRef) => {
  const theme = React.useContext(ThemeContext)

  const { leftAddon, rightAddon } = useAddons({
    Component: ButtonAddon,
    leftAddon: props.leftAddon,
    rightAddon: props.rightAddon,
    props: {
      hasHover: props.hasHover,
      hasWaveEffect: props.hasWaveEffect,
      isOutline: props.isOutline,
      isSoft: props.isSoft,
      palette: props.palette,
      shape: props.shape,
      size: props.size,
      sizeSm: props.sizeSm,
      sizeMd: props.sizeMd,
      sizeLg: props.sizeLg,
      sizeXl: props.sizeXl,
      waveEffectPalette: props.waveEffectPalette,
    },
  })

  const waveEffectPalette = React.useMemo(() => getWaveEffectPalette(props, theme.$.vars.isDark), [props.palette, props.isOutline, props.isSoft, props.isLink, props.waveEffectPalette])

  const getLoadingEl = () => {
    if (!props.loading) {
      return null
    }

    const loadingProps = {
      isOutline: props.isOutline,
      isSoft: props.isSoft,
      palette: props.palette,
      status: props.isLoading,
    }

    if (props.loading.type?.displayName === 'SRCLoading') {
      return React.cloneElement(props.loading, loadingProps)
    }

    return <ButtonLoading {...loadingProps}>{props.loading}</ButtonLoading>
  }

  const [loadingEl, setLoadingEl] = React.useState<Nullable<JSX.Element>>(() => getLoadingEl())

  useChangeEffect(() => {
    setLoadingEl(getLoadingEl())
  }, [props.loading, props.isLoading])

  const applyIntrinsicStyledCoreProps = () => ({
    ...extractElementProps(props, [intrinsicStyledCoreProps]),
    ...(props.hasSpace && {
      margin: `$size.button.${props.size}.margin.y $size.button.${props.size}.margin.x`,
      ...(props.sizeSm && { marginSm: `$size.button.${props.sizeSm}.margin.y $size.button.${props.sizeSm}.margin.x` }),
      ...(props.sizeMd && { marginMd: `$size.button.${props.sizeMd}.margin.y $size.button.${props.sizeMd}.margin.x` }),
      ...(props.sizeLg && { marginLg: `$size.button.${props.sizeLg}.margin.y $size.button.${props.sizeLg}.margin.x` }),
      ...(props.sizeXl && { marginXl: `$size.button.${props.sizeXl}.margin.y $size.button.${props.sizeXl}.margin.x` }),
    }),
  })

  const hasButtonContainer = leftAddon?.props?.isSeparated || rightAddon?.props?.isSeparated

  let Element = (
    <ButtonElement
      {...props.elementProps}
      {...((!props.isDisabled && !props.isLoading) && extractElementProps(props, [clickEvents]))}
      {...(!hasButtonContainer && applyIntrinsicStyledCoreProps())}
      {...(props.as && { as: props.as })}
      buttonSize={props.size}
      buttonSizeSm={props.sizeSm}
      buttonSizeMd={props.sizeMd}
      buttonSizeLg={props.sizeLg}
      buttonSizeXl={props.sizeXl}
      hasButtonContainer={hasButtonContainer}
      hasHover={props.hasHover}
      hasLeftAddon={!!leftAddon}
      hasRightAddon={!!rightAddon}
      hasSeparatedLeftAddon={!!leftAddon?.props?.isSeparated}
      hasSeparatedRightAddon={!!rightAddon?.props?.isSeparated}
      isActive={props.isActive}
      isBlock={props.isBlock}
      isDisabled={props.isDisabled}
      isFixedSize={props.isFixedSize}
      isLink={props.isLink}
      isLoading={props.isLoading}
      isSoft={props.isSoft}
      isOutline={props.isOutline}
      palette={props.palette}
      ref={forwardRef}
      shape={props.shape}
    >
      { loadingEl && loadingEl }
      { (leftAddon && !leftAddon.props.isSeparated) && leftAddon }
      <ButtonContent>{props.children}</ButtonContent>
      { (rightAddon && !rightAddon.props.isSeparated) && rightAddon }
    </ButtonElement>
  )

  if (props.hasWaveEffect) {
    Element = <WaveEffect palette={waveEffectPalette}>{Element}</WaveEffect>
  }

  if (hasButtonContainer) {
    Element = (
      <Div
        {...applyIntrinsicStyledCoreProps()}
        display={props.isBlock ? 'flex' : 'inline-flex'}
      >
        { leftAddon?.props?.isSeparated && leftAddon }
        { Element }
        { rightAddon?.props?.isSeparated && rightAddon }
      </Div>
    )
  }

  return Element
})

Button.defaultProps = {
  elementProps: {},
  hasHover: true,
  hasWaveEffect: true,
  palette: 'primary',
  shape: 'rectangle',
  size: 'medium',
  type: ButtonType.BUTTON,
}

export default Button
