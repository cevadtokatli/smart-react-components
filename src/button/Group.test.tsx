import React from "react"
import renderTheme from "../test/renderTheme"
import Group from "./Group"
import Button from "./Button"

describe("<Group />", () => {
    it("should match snapshot", () => {
        const wrapper = renderTheme(
            <Group>
                <Button>Button-1</Button>
                <Button>Button-2</Button>
                <Button>Button-3</Button>
            </Group>
        )
        expect(wrapper).toMatchSnapshot()
    })
})