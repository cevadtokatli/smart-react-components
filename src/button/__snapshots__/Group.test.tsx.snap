// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Group /> should match snapshot 1`] = `
<div
  class="sc-AxjAm sc-AxirZ goZcrM"
  display="inline-block"
>
  <a
    class="sc-AxhCb sc-AxhUy iPdFHO"
    data-src-wave-effect="0"
    shape="default"
    tabindex="0"
  >
    Button-1
  </a>
  <a
    class="sc-AxhCb sc-AxhUy iPdFHO"
    data-src-wave-effect="0"
    shape="default"
    tabindex="0"
  >
    Button-2
  </a>
  <a
    class="sc-AxhCb sc-AxhUy iPdFHO"
    data-src-wave-effect="0"
    shape="default"
    tabindex="0"
  >
    Button-3
  </a>
</div>
`;
